src/query/block/lazy.go

```diff
@@ -83,12 +83,20 @@ func (it *lazyStepIter) Meta() Metadata {
 }
 
 func (it *lazyStepIter) Current() Step {
-	c := it.it.Current()
-	tt := it.opts.TimeTransform()
+	var (
+		c        = it.it.Current()
+		tt, vt   = it.opts.TimeTransform(), it.opts.ValueTransform()
+		stepVals = c.Values()
+	)
+
+	vals := make([]float64, 0, len(stepVals))
```

Comment:
longer term, instead of allocating for each `Current()` call, you can reuse a field in the struct. it'd look something like this:

```
type lazyStepIter struct {
  it   StepIter		
  opts LazyOptions
  values []float64
}

func (it *lazyStepIter) Current() Step {
var (
	c        = it.it.Current()
	tt, vt   = it.opts.TimeTransform(), it.opts.ValueTransform()
	stepVals = c.Values()
)
it.values = it.values[:0]
if cap(it.values) < len(stepVals) {
  it.values = make([]float64, 0, len(stepVals))
}
...
  return ColStep{ 
    ...
    values: it.values,
  }
}
```

note: this might cause issues if the callers of the this function are relying on the returned value from this function to be valid after multiple calls to Next()/Current(). Dunno how it's used so can't say for sure. Maybe file an issue to look into this

Reply:
If we do it this way, would make sense to move all this logic into the `Next` function so we don't have to recalculate :+1:

Reply:
Yeah, Next() would definitely be the better place for it.